password.go

```diff
@@ -17,9 +20,49 @@ var PasswordQuestionTemplate = `
 {{- color "green+hb"}}? {{color "reset"}}
 {{- color "default+hb"}}{{ .Message }} {{color "reset"}}`
 
+type filteredReadlinePasswordOutput struct {
+	len int
+}
+
+// Write will be called by readline to echo the masked characters to the console.  It assumes
+// full control of the output with prompting, but since survey is printing the prompt readline
+// does not play nice here.  Write is called with clear-line statements which need to be
+// ignored then it is called with the full buffered output. So if you type "hi" this is the
+// sequence that will be sent to Write
+// 0: "\033[J\033[2K\r"
+// 1: " \b"
+// 2: "\033[J\033[2K\r"
+// 3: "*" (this is the masked "h" from "hi")
+// 4: "\033[J\033[2K\r"
+// 5: "**" (this is the masked "hi" from "hi")
+func (o *filteredReadlinePasswordOutput) Write(p []byte) (n int, err error) {
```

Comment:
Yea it's probably better to just leave it up to the prompt to "decide" if it wants to support help text. 

I'm definitely interested in the feature, just want to make sure we think through all the possibilities so we don't end up having to bump a major version because we missed something obvious. I suppose it fits in the same way `Message` is defined on the prompt. Although, I might be able to see an argument to move that to the question type too, but let's not go too far down that rabbit hole just yet.

As a side note, I think it would be good to show some help text to describe the arrow key behaviour of the multiline inputs for exactly this reason - was planning on adding it soon